### ISSUE TITLE
Reporting what ip my peers think I am at

### ISSUE THREAD

**Issue by ChristopherA** [ipfs/go-ipfs#1147]
Again, related to debugging peers in odd network situations, it would be useful to collect what ip addresses my peers are using to connect to me. This way I can debug nat traversal, routing differences, etc.

**Comment by jbenet** [comment]
@ChristopherA for now, you can view your addresses with:

```sh
ipfs id
# Addresses. addresses that are learned by peers and told to us will show up here.
```
and
```
ipfs swarm addrs
```
these are all the addresses the node is currently aware of.

**Comment by ChristopherA** [comment]
Yes, I see now that `ipfs id` does show what I'm looking for. I think it would be useful if `ipfs swarm addrs` also showed the addresses for the home daemon so that they are all in one place.

**Comment by jbenet** [comment]
btw, behind symmetric nats, it's sometimes impossible to get a listenable addr. Our nat-traversal isn't great yet, will get better with time.

> I think it would be useful if `ipfs swarm addrs` also showed the addresses for the home daemon so that they are all in one place.

Yeah, agreed. relevant change: add own addrs to https://github.com/ipfs/go-ipfs/blob/master/core/commands/swarm.go#L107-L116

**Comment by whyrusleeping** [comment]
TODO: add own addresses to ipfs swarm addrs output
